platform/core/src/services/HangingProtocolService/ProtocolEngine.js

```diff
@@ -62,33 +70,35 @@ export default class ProtocolEngine {
 
       // If it is not already in the MatchedProtocols Collection, insert it with its score
       if (!this.matchedProtocols.has(protocol.id)) {
-        console.log(
-          'ProtocolEngine::updateProtocolMatches inserting protocol match',
-          matchedDetail
-        );
+        // console.log(
+        //   'ProtocolEngine::updateProtocolMatches inserting protocol match',
+        //   matchedDetail
+        // );
         this.matchedProtocols.set(protocol.id, protocol);
         this.matchedProtocolScores[protocol.id] = matchedDetail.score;
       }
     });
   }
 
-  findMatch(metaData, rules) {
+  findMatch(metaData, rules, extraMatchData) {
     return HPMatcher.match(
       metaData,
       rules,
-      this.customAttributeRetrievalCallbacks
+      this.customAttributeRetrievalCallbacks,
+      extraMatchData,
     );
   }
 
   /**
    * Finds the best protocols from Protocol Store, matching each protocol matching rules
    * with the given study. The best protocol are ordered by score and returned in an array
    * @param  {Object} study StudyMetadata instance object
+   * @param {object} extraMatchData to match on.
    * @return {Array}       Array of match objects or an empty array if no match was found
    *                       Each match object has the score of the matching and the matched
    *                       protocol
    */
-  findMatchByStudy(study) {
+  findMatchByStudy(study, extraMatchData) {
```

Comment:
I would rename `extraMatchData` to `options` or something similar, there is no need to have `Match` in the name since the method name is `findMatchByStudy`